server/src/main/java/org/candlepin/pinsetter/core/PinsetterKernel.java

```diff
@@ -170,6 +171,22 @@ else if (!isClustered()) {
             log.debug("Jobs implemented:" + jobImpls);
             Set<JobKey> jobKeys = scheduler.getJobKeys(jobGroupEquals(CRON_GROUP));
 
+            /*
+             * purge jobs that have been deleted from this version of Candlepin.
+             * This is necessary as we might not even have the Class definition
+             * at classpath, Hence any attempt at fetching the JobDetail by the
+             * Scheduler or JobStatus by the JobCurator will fail.
+             */
```

Comment:
Could we provide/inject a replacement class for deleted job classes?  

Something like a base 'DeletedTask', and when the job goes away, we make say StatisticsHistoryTask a sub class of DeletedTask. That way the jobkey->task class reference doesn't break, but we could tell it's an existing job that can be ignored/deleted.